api/rosetta/data_balance_integration_test.go

```diff
@@ -90,69 +104,79 @@ func TestGetBalance(t *testing.T) {
+			wantBalances:  []string{"10000099999"},
+			validateBlock: validateBlockID(t, second.Index, second.Hash),
 		},
 		{
-			name:           "valid balance request - last indexed block",
-			request:        balanceRequest("754aed9de6197641", 425, "594d59b2e61bb18b149ffaac2b27b0efe1854f6795cd3bb96a443c3676d78683"),
-			wantBalance:    "10000100002",
-			wantStatusCode: http.StatusOK,
-			wantHandlerErr: assert.NoError,
+			name:          "last indexed block",
+			request:       balanceRequest(testAccount, last),
+			wantBalances:  []string{"10000100002"},
+			validateBlock: validateBlockID(t, last.Index, last.Hash),
 		},
 		{
-			name:           "empty balance request",
-			request:        rosetta.BalanceRequest{},
-			wantStatusCode: http.StatusUnprocessableEntity,
-			wantHandlerErr: assert.Error,
-		},
-		/*
-			TODO: when we re-add block validation, we should reintroduce this test case - block height and hash mismatch should result in an error
-				=> https://github.com/optakt/flow-dps/issues/51
-			{
-				name:           "invalid request - block hash and height mismatch",
-				request:        balanceRequest("754aed9de6197641", 50, "594d59b2e61bb18b149ffaac2b27b0efe1854f6795cd3bb96a443c3676d78683"),
-				wantStatusCode: http.StatusUnprocessableEntity,
-				wantHandlerErr: assert.Error,
-			},
-		*/
-		{
-			name:           "invalid account address",
-			request:        balanceRequest("invalid_address", 0, "d47b1bf7f37e192cf83d2bee3f6332b0d9b15c0aa7660d1e5322ea964667b333"),
-			wantStatusCode: http.StatusUnprocessableEntity,
-			wantHandlerErr: assert.Error,
+			// use block height only to retrieve data, but verify hash is set in the response
+			name:          "get block via height only",
+			request:       balanceRequest(testAccount, identifier.Block{Index: second.Index}),
+			wantBalances:  []string{"10000099999"},
+			validateBlock: validateBlockID(t, second.Index, second.Hash),
 		},
 		{
-			name:           "invalid balance request - account does not exist yet",
-			request:        balanceRequest("754aed9de6197641", 12, "9035c558379b208eba11130c928537fe50ad93cdee314980fccb695aa31df7fc"),
-			wantStatusCode: http.StatusInternalServerError,
-			wantHandlerErr: assert.Error,
-		},
-		{
-			name:           "invalid balance request - unknown block",
-			request:        balanceRequest("754aed9de6197641", 426, "xyz"),
-			wantStatusCode: http.StatusInternalServerError,
-			wantHandlerErr: assert.Error,
+			// we currently have only the one token, so we can't really request different currencies
+			name: "get multiple currencies",
+			request: rosetta.BalanceRequest{
+				NetworkID: defaultNetwork(),
+				AccountID: identifier.Account{Address: testAccount},
+				BlockID:   last,
+				Currencies: []identifier.Currency{
+					{Symbol: dps.FlowSymbol, Decimals: dps.FlowDecimals},
+					{Symbol: dps.FlowSymbol, Decimals: dps.FlowDecimals},
+					{Symbol: dps.FlowSymbol, Decimals: dps.FlowDecimals},
+				},
```

Comment:
We should probably not test this; semantically, this will work, but it really shouldn't be used. I would abandon that test case until we care about more than one currency.